sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -946,6 +947,38 @@ object functions {
   def cosh(columnName: String): Column = cosh(Column(columnName))
 
   /**
+   * Adds a number of days to startDate, given values for startDate and days
+   *
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def date_add(startDate: Column, days: Column): Column = DateAdd(startDate.expr, days.expr)
+
+  /**
+   * Adds a number of days to startDate, given column names for startDate and days
+   *
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def date_add(startDate: String, days: String): Column = date_add(Column(startDate), Column(days))
```

Comment:
Good point, we have many functions which take two or more parameters, all the parameters could Column or Literal, we usually take only a few combination of them as DataFrame function. It's hard to have a rule to follow, but we usually take the first one as Column, and others as Literal.

There are a few cases that we have all the combinations (for example, atan2, pow), that's bad, we should revisit them, cc @rxin @brkyvz